### DISCUSSION TITLE
upper/lower-case filters for mime-version and message-id

### DISCUSSION THREAD

**Discussion by kkretsch** [rspamd/rspamd#3923] [Q&A]
Even if it only gives 0.5 negative spam points, how valid is the decision to give **Mime-Version** and worse ranking than **MIME-Version**? And the same goes to Message-Id versus Message-ID.

First question would be: Which RFC exactly defines the _uppercasyness_ of mail headers?

And second: How to solve the fight between OSS platforms not having the same understanding of it?

One sends the mail and defines that as _the-right-thing_, the other filters them as _common practice_.

**Comment by kkretsch** [comment]
Topic started here:
https://meta.discourse.org/t/rspamd-complains-about-mime-version-header/205329

**Comment by vstakhov** [comment]
Rspamd marks something that is commonly used in spam with some small valued positive score. I see no issues in this behaviour - Rspamd is apparently not an RFC validator.

**Reply by kkretsch** [reply]
So for you it is more the _common-practice_ decision. Is the any statistical numbers how many MIME vs Mime is used in SPAM vs. HAM?
Or how did that rule get invented?

**Reply by alexvb** [reply]
Just noticed that Apple Mail also uses the worse ranking casing `Mime-Version`.

**Comment by tmlmt** [comment]
As explained [here](https://stackoverflow.com/a/6143644), email headers are not case-sensitive according to the spec. As noted by @alexvb just above, some email apps like Apple Mail use the worst ranking casing of `Mime-Version`. I would tend to disable those case checking rules. Should they even be removed from rspamd?
